src/brod_client.erl

```diff
@@ -414,7 +415,15 @@ handle_call({get_transactional_coordinator, TransactionId}, _From, State) ->
   {Result, NewState} = do_get_transactional_coordinator(State, TransactionId),
   {reply, Result, NewState};
 handle_call({start_producer, TopicName, ProducerConfig}, _From, State) ->
-  {Reply, NewState} = do_start_producer(TopicName, ProducerConfig, State),
+  %% NOTE: Store ProducerConfig into the config in state
```

Comment:
producer config is per topic, it is already stored in supervisor child spec.
we can port get_childspec to brod_supervisor3

https://github.com/erlang/otp/blob/ec6323dd5065203737b0a48d074f116f0c1a2719/lib/stdlib/src/supervisor.erl#L752-L759

Reply:
Done.